airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/spec.json

```diff
@@ -4,27 +4,69 @@
     "$schema": "http://json-schema.org/draft-07/schema#",
     "title": "SurveyMonkey Spec",
     "type": "object",
-    "required": ["access_token", "start_date"],
+    "required": ["start_date"],
```

Comment:
Isn't `credentials` required field?

Reply:
we don't need to add `credentials`, `client_secret` and `client_id` to required fields, because it will break connection for old users.

Reply:
You're right but I think we need to make an arbitrage there: 
1. Break the old configurations but have a clear specification
2. Do not require this field, which can lead to multiple configuration errors we'll have to manage downstream.
@YowanR can you please suggest what to do in this scenario?

Reply:
Option B will allow users not to fill the `credentials` field, leading to a failed connection check. 
This will produce on-call Sentry errors. 
A workaround would be to add a new logic to the `check`: Make the check return `false` if these value are empty. This will handle this case gracefully and not create OC issues. Wdyt @darynaishchenko / @girarda ?

Reply:
@alafanechere for me looks good, I can add this logic to the `check`

Reply:
I'm fine with option B.

@YowanR for context: the downside of Option is B is we need to maintain logic for every breaking change. The idea situation would be to support [config migration](https://github.com/airbytehq/airbyte/issues/5396)

Reply:
There was an update to the config migration EPIC. Is that sufficient to support the better option now?

Reply:
@alafanechere added credential check, please take a look